### ISSUE TITLE
systemd service fails if deployment in additionalManifests fails

### ISSUE THREAD

**Issue by agracey** [k3s-io/k3s#11380]
**Environmental Info:**
K3s Version: 
v1.30.3+ks31

Node(s) CPU architecture, OS, and Version: 
x86_64, SL Micro 6.0

Cluster Configuration:

Single node (but shouldn't matter)

**Describe the bug:**

If I create a broken deployment in `/var/lib/rancher/k3s/server/manifests` then install k3s, the `k3s-server` systemd service will go into a failed state and nothing that depends on it will start. 

**Expected behavior:**
The k3s-server service should be considered running even if a custom addOn fails.

**Actual behavior:**
The k3s-server service fails. 

**Additional context / logs:**
I'm using Edge Image Builder to build an appliance image.

**Comment by brandond** [comment]
> If I create a broken deployment in /var/lib/rancher/k3s/server/manifests then install k3s, the k3s-server systemd service will go into a failed state and nothing that depends on it will start.

No, it doesn't. Did you perhaps deploy something that is so broken it causes other core components to fail? The invalid manifest itself will not cause k3s to exit.

Please post the full logs showing what you're claiming.
